### PR TITLE
feat: Display custom WelcomeScreen when OnboardingPartner context found

### PR DESCRIPTION
**Pull request by Ldoppea**
On App's first launch, we want to detect the installation context so we
can extract Onboarding Partner and display custom instructions to our
users

To do this we use Google Play Install Referrer API, and so it is
enabled only on Android phones

This PR only displays the raw OnboardingPartner info (if found)
Real screen implementation will be done later

### CODE REVIEW

File: src/screens/welcome/install-referrer/androidPlayInstallReferrer.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+      } as PlayInstallReferrerInfo)
+    }
+
+    return new Promise(resolve => {
+      PlayInstallReferrer.getInstallReferrerInfo(
+        (installReferrerInfo, error): void => {
+          if (!error) {
+            resolve(installReferrerInfo)
+          } else {
+            // Here we silence the error as non-google devices would trigger it
```
Comment (Ldoppea) [inline src/screens/welcome/install-referrer/androidPlayInstallReferrer.ts:30]: I would appreciate feedback on this specific choice I made (silencing error and sentry). It that choice OK for you?

File: src/screens/welcome/WelcomeScreen.jsx [hunk 2]
```diff
@@ -8,6 +8,7 @@ import { makeHandlers } from '/libs/functions/makeHandlers'
 import { getColors } from '/ui/colors'
 import { useDimensions } from '/libs/dimensions'
 import { LoginScreen } from '/screens/login/LoginScreen'
+import { useInstallReferrer } from './install-referrer/useInstallReferrer'
```
Comment (Crash--) [inline src/screens/welcome/WelcomeScreen.jsx:11]: alias?

File: src/screens/welcome/install-referrer/onboardingPartner.ts [hunk 3]
```diff
@@ -0,0 +1,127 @@
+import AsyncStorage from '@react-native-async-storage/async-storage'
+import { Platform } from 'react-native'
+
+import Minilog from '@cozy/minilog'
+
+import { getInstallReferrer } from './androidPlayInstallReferrer'
```
Comment (Crash--) [inline src/screens/welcome/install-referrer/onboardingPartner.ts:6]: alias?

File: src/screens/welcome/install-referrer/onboardingPartner.ts [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+}
+
+export type OnboardingPartner = WithOnboardingPartner | NoOnboardingPartner
+
+const noOnboardingPartner = (): NoOnboardingPartner => ({ hasReferral: false })
+
+const extractOnboardingPartner = (
+  installReferrer: string
+): OnboardingPartner | null => {
+  const url = new URL(`https://cozy.io?${installReferrer}`)
```
Comment (Crash--) [inline src/screens/welcome/install-referrer/onboardingPartner.ts:34]: I don't understand this domain? Don't we have to only generate links as : play.google.com/..?utm_... ? What is this `cozy.io` stuff?

File: src/screens/welcome/install-referrer/onboardingPartner.ts [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+    )
+
+    return onboardingPartner
+  }
+
+export const getOnboardingPartner = async (): Promise<OnboardingPartner> => {
+  if (Platform.OS !== 'android') return noOnboardingPartner()
+
+  const onboardingPartnerFromAsyncStorage =
+    await getOnboardingPartnerFromAsyncStorage()
```
Comment (Crash--) [inline src/screens/welcome/install-referrer/onboardingPartner.ts:115]: So you write the information to the async storage to be able to re use it later if the user is not logged, right? 

When are you removing the info? Never?

File: docs/how-to-debug-onboarding-partner.md [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+- Campaign Content: the provided context (i.e `MesPapiers`, `default` etc)
+- Campaign Name: `onboarding_partner`
+
+## Testing on production (phone with Play Store installed)
+
+When testing on production, build a referrer URI and install the App by opening this URI from your device
+
+## Testing on local dev (phone with Play Store installed)
+
+If the feature you want to test is not deployed yet, the do the following:
```
Comment (Crash--) [inline docs/how-to-debug-onboarding-partner.md:27]: then do the

File: src/screens/welcome/install-referrer/onboardingPartner.ts (hunk 4 above)
Comment (Ldoppea) [reply src/screens/welcome/install-referrer/onboardingPartner.ts:34]: The used API returns only the query string in the form of `utm_source=SOME_SOURCE&utm_content=SOME_CONTEXT&utm_campaign=onboarding_partner&anid=admob`

The `cozy.io` part is just some whatever domain so I can create an URL (I don't know how to create a `new URLSearchParams` from a string)

File: src/screens/welcome/install-referrer/onboardingPartner.ts (hunk 5 above)
Comment (Ldoppea) [reply src/screens/welcome/install-referrer/onboardingPartner.ts:115]: Never, we chose to display the same contextualized WelcomeScreen + LoginScreen even if the user logs out

**Review by zatteo — APPROVED** [review]

File: src/screens/welcome/WelcomeScreen.jsx (hunk 2 above)
Comment (Ldoppea) [reply src/screens/welcome/WelcomeScreen.jsx]: Fixed in https://github.com/cozy/cozy-flagship-app/compare/e296534f408c5ca5dce5b8af44df9d6144165350..1b2e03a8f035457424a35b1ed39ad6fbb317bb73

File: src/screens/welcome/install-referrer/onboardingPartner.ts (hunk 3 above)
Comment (Ldoppea) [reply src/screens/welcome/install-referrer/onboardingPartner.ts]: Fixed in https://github.com/cozy/cozy-flagship-app/compare/e296534f408c5ca5dce5b8af44df9d6144165350..1b2e03a8f035457424a35b1ed39ad6fbb317bb73

File: docs/how-to-debug-onboarding-partner.md (hunk 6 above)
Comment (Ldoppea) [reply docs/how-to-debug-onboarding-partner.md]: Fixed in https://github.com/cozy/cozy-flagship-app/compare/e296534f408c5ca5dce5b8af44df9d6144165350..1b2e03a8f035457424a35b1ed39ad6fbb317bb73

**Pull request merged**